app/tests/challenges_tests/test_permissions.py

```diff
@@ -76,3 +79,59 @@ def _get_view():
             assert response.status_code == 200
 
             remove_perm(permission, u, obj)
+
+
+@pytest.mark.django_db
+def test_request_challenge_only_when_verified(client):
+    user = UserFactory()
+    assert not Verification.objects.filter(user=user)
+
+    response = get_view_for_user(
```

Comment:
Now we have some time to look at the tests It is better to test the view directly rather than using get_view_for_user. Have you looked at Adam Johnsons book on Django testing?